### ISSUE TITLE
Retrieving User Info with initial returned ID token

### ISSUE THREAD

**Issue by yashrao99** [okta/okta-oidc-ios#204]
Hi there,

The JWT that is returned to me with the `OktaOidcStateManager` contains some of the relevant information I want, but requires the iOS client to use the `getUser` method to retrieve other information. 

I need that information within the JWT that is returned from the authentication. I cannot add these values to the JWT after that call, or else that would defeat the purpose. 

I have tried configuring the profile mapping to return those values in the original JWT. Our web-applications profile-mapping for these very same values works, and can be found in the original JWT, but for our mobile clients, the JWT does not return the configured profile mapping values we are expecting.

Any help with this would be much appreciated!

Yashas Rao

**Comment by IldarAbdullin-okta** [comment]
Hi @yashrao99 ,

Could you try to add more scopes into your `Okta.plist` and check? Add the following scopes: `profile`, `email` and `phone`

Ildar

**Comment by yashrao99** [comment]
@IldarAbdullin-okta Hi there,

Adding those to the scope did not work in retrieving the information in the `JWT` with the relevant information. Still having to hit the `userInfo` endpoint

**Comment by yashrao99** [comment]
@IldarAbdullin-okta Hi there,

We tried again today, by adding all the values to the scope, and we are still not getting the personal claims that we expected. We have the following `openid profile offline_access email phone`

Thanks,
Yashas Rao

**Comment by IldarAbdullin-okta** [comment]
@yashrao99 , have you tried to hit `userinfo` endpoint and check whether you see required information in response?
Please also read this post - https://devforum.okta.com/t/how-to-get-more-claims-in-userinfo-endpoint/3559

**Comment by yashrao99** [comment]
@ildarAbdullin-okta

I have used the userInfo endpoint and it does work. My question is that our
web applications JWT contains the relevant info when decided.

However, using the mobile SDK, I have to fetch that information using
userInfo. I would like that information in the original JWT or idToken that
is returned rather than having to make an extra trip

**Comment by IldarAbdullin-okta** [comment]
I assume you mean `id_token` when saying JWT. Could you check link from Okta devforum that I provided? It has some relevant information about how to add custom claims into `id_token`
I would also recommend to contact Okta customer support

**Comment by yashrao99** [comment]
Yes, `id_token` == JWT. On our web app, decoding this `id_token` contains the values we configured using the profile mappings but using the mobile SDK, we are required to use the `access_token` to hit the `/userinfo` endpoint which we are trying to avoid since we need our custom values within the returned `id_token`

I checked that link. The only one that looked relevant was referencing a `Claims` tab from the admin console. I am looking at my admin console, I am able to configure profile mappings, which I already have.
